### PR TITLE
Alvin 'PR

### PR DESCRIPTION
**Pull request by HoiDam**

### CODE REVIEW

**Comment by mariechow-oo** [comment]
Well:
- can add todo and display correctly
- have line through when toggle todo status
- can delete

Less well:
- can have more commits
- delete and toggle not working correctly

File: src/features/TodoGenerator.js [hunk 1]
```diff
@@ -1,12 +1,26 @@
...
   const onInputChange = (event) => {
     setTodo(event.target.value);
   };
+
+  const dispatch = useDispatch();
   const onAdd = () => {
-    props.addTodo(todo);
+    dispatch(addToDo(
+      {
+        "id:":uuid(),
```
Comment (mariechow-oo) [inline src/features/TodoGenerator.js]: extra :

File: src/features/TodoGenerator.js [hunk 2]
```diff
@@ -1,12 +1,26 @@
...
+  
   const onInputChange = (event) => {
     setTodo(event.target.value);
   };
+
+  const dispatch = useDispatch();
   const onAdd = () => {
-    props.addTodo(todo);
+    dispatch(addToDo(
+      {
```
Comment (mariechow-oo) [inline src/features/TodoGenerator.js:16]: can put this in TodoSlice

File: src/features/ToDoSlice.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    name:"todo",
+    initialState:{
+        todoList:[],
+    },
+    reducers:{
+        addToDo: (state, action)=>{
+            state.todoList.push(action.payload)
+        },
+        doneToDo: (state,action)=>{
+            state.todoList.splice(action.payload,1)
```
Comment (mariechow-oo) [inline src/features/ToDoSlice.js]: should use the index instead of action.payload (which is id) when splice

**Comment by HoiDam** [comment]
The issue should be fixed finally =]

**Pull request closed without merging**